server/src/test/java/org/elasticsearch/cluster/NodeConnectionsServiceTests.java

```diff
@@ -400,10 +396,8 @@ public void testDebugLogging() {
 
             runTasksUntil(deterministicTaskQueue, CLUSTER_NODE_RECONNECT_INTERVAL_SETTING.get(Settings.EMPTY).millis());
             appender.assertAllExpectationsMatched();
-        } finally {
```

Comment:
Do you think we can check that `assertAllExpectationsMatched` has been called at least once before closing the captured releasable?

Reply:
Yes that makes sense - indeed we should have called `assertMatched` on every registered expectation. I won't be doing that here but a followup would be welcome.

Reply:
I can take this as a follow up, no need for you to spend time on this now.

Reply:
I opened #92765